crates/krun/src/cli_options.rs

```diff
@@ -29,6 +30,22 @@ pub fn options() -> OptionParser<Options> {
             None => Ok((s, None)),
         })
         .many();
+    let mem = long("mem")
+        .help(
+            "The amount of RAM, in MiB, that will be available to this microVM.
+            The memory configured for the microVM will not be reserved
+            immediately. Instead, it will be provided as the guest demands it,
+            and both the guest and libkrun (acting as the Virtual Machine
+            Monitor) will attempt to return as many pages as possible to the
+            host.
+    [default: 80% of total RAM]",
+        )
+        .argument("MEM")
+        .guard(
+            |&mem| mem <= MiB::from(16384),
+            "the maximum amount of RAM supported is 16384 MiB",
```

Comment:
(Moved from https://github.com/teohhanhui/krun/commit/ce29e32992389569b70076ccc791577b8d0cfe28#r142414630)

@asahilina
> Is there a reason for this limit? We have machines with 192GB of RAM, surely it makes sense for the VM to support more than 16G...

I have no idea why there's this limit in krunvm. Was going to ask @slp since he's the original author of that code according to git blame. Will remove once we confirm that it's not needed.

Reply:
Originally, libkrunfw's kernel was limited to 8 vCPU and 16GB to reduce the kernel size, since it was intended to be used for container isolation exclusively. Since we've now incorporated other use cases, we should lift that limit. It needs to be done in libkrun and libkrunfw first, so it makes sense to keep that limit in krun in the meantime.